m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+                                    LOAN TRANSACTION:POSTDATE>=TARGETSTARTDATE AND
+                                   (LOAN TRANSACTION:POSTDATE<=TARGETENDDATE OR
+                                    TARGETENDDATE=DATENULL) AND
+                                    LOAN TRANSACTION:VOIDCODE=0 AND
+                                    LOAN TRANSACTION:USERNUMBER=TARGETSYMXUSER)
+    DO
+     IF LOAN TRANSACTION:COMMENTCODE=1 THEN
+      DO
+       IF TCCOUNT<TCCOUNTMAX THEN [store pertinant data for future use]
+        DO
+         TCCOUNT=TCCOUNT+1
+         TCCOMMENT(TCCOUNT)=LOAN TRANSACTION:COMMENT
+         TCIDTYPE(TCCOUNT)=LOAN TRANSACTION:IDTYPE
+         TCID(TCCOUNT)=LOAN TRANSACTION:ID
+         TCSTMTDESCR(TCCOUNT)=LOAN TRANSACTION:STMTDESC:1
+        END
+      END
+     ELSE [transaction is not a comment - validate & process]
+      DO
+       IF TCCOUNT=1 THEN [transaction with statement description]
+        DO
+         IF CHARACTERSEARCH(LOAN TRANSACTION:STMTDESC:1,TARGETCOMMENT)>0 THEN
+          DO [valid target transaction - add to report]
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=LOAN TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END
+        END [IF TCCOUNT=1 THEN...]
+       ELSE IF TCCOUNT=2 THEN [transaction with tran comment]
+        DO
+         IF CHARACTERSEARCH(TCCOMMENT(TCCOUNT-1),TARGETCOMMENT)>0 THEN
+          DO
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=LOAN TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END [ELSE IF TCCOUNT=2 THEN...]
+        END [IF TCCOUNT=2]
+      END [LOAN TRANSACTION:COMMENTCODE<>1]
+    END [FOR EACH LOAN TRANSACTION WITH...]
+  END [FOR EACH LOAN]
+END [PROCEDURE]
+
+PROCEDURE PRINTTRANTOREPORT
+[* Print the detail for an immediate transfer (from tran history)
+*]
+ COL=001 ACCOUNT:NUMBER
+ COL=013 TRANSOURCECHR(TRANSOURCE)
+ COL=020 TRANSOURCEID
+ COL=026 TOACCOUNT
+ COL=038 TOSLIDTYPE
+ COL=045 TOSLID
+ COL=061 RIGHT TOTRANAMOUNT
+ COL=064 "IMMEDIATE"
+ NEWLINE
+END
+
+PROCEDURE CLEARTRANVARS
+[* Clear the variables storing transactional detail
+*]
+ TCCOUNT=0
+ FOR TMPLOOP=1 TO TCCOUNT
```

Comment:
I recommend change TO to TCCOUNTMAX instead of TCCOUNT in the FOR Loop, as TCCOUNT is 0 at that point, and also incase you start using the for loop for variable initialization.

Reply:
Changed the order instead. Left TCCOUNT as the loop limiter so Robby wouldn't call me out on it :-)

Reply:
@rrobison44 Works for me! Done